TextInputKit/TextInputKit/Code/SpecializedTextInput/CardNumber/BankCardNumberUtils.swift

```diff
@@ -10,119 +10,64 @@ import Foundation
 
 struct BankCardNumberUtils {
 
-    struct IinRangeInfo {
-
-        let range: CountableClosedRange<Int>
-
-        let cardBrand: BankCardBrand
-
-        let maxCardNumberLength: Int
+    private init() {}
 
-        fileprivate init(_ range: CountableClosedRange<Int>, _ cardBrand: BankCardBrand, _ maxCardNumberLength: Int) {
-            self.range = range
-            self.cardBrand = cardBrand
-            self.maxCardNumberLength = maxCardNumberLength
-        }
+}
 
-    }
+extension BankCardNumberUtils {
 
     static func cardNumberDigitsStringView(
         from stringView: String.UnicodeScalarView
         ) -> String.UnicodeScalarView? {
 
-        return self.stringView(
+        return StringUtils.stringView(
             from: stringView,
             preservingCharacters: CharacterSet.decimalDigits,
-            ignoringCharacters: CharacterSet.whitespaces)
+            ignoringCharacters: cardNumberWhitespaces)
     }
 
     static func cardNumberDigitsStringViewAndRange(
         from stringView: String.UnicodeScalarView,
         range: Range<String.UnicodeScalarIndex>
         ) -> (String.UnicodeScalarView, Range<String.UnicodeScalarIndex>)? {
 
-        return self.stringViewAndRange(
+        return StringUtils.stringViewAndRange(
             from: stringView,
             range: range,
             preservingCharacters: CharacterSet.decimalDigits,
-            ignoringCharacters: CharacterSet.whitespaces)
+            ignoringCharacters: cardNumberWhitespaces)
     }
 
-    static func stringView(
-        from stringView: String.UnicodeScalarView,
-        preservingCharacters: CharacterSet,
-        ignoringCharacters: CharacterSet
-        ) -> String.UnicodeScalarView? {
+    private static let cardNumberWhitespaces = CharacterSet(charactersIn: " ")
 
-        do {
-            var resultStringView = "".unicodeScalars
-            try scan(stringView,
-                preservingCharacters: preservingCharacters,
-                ignoringCharacters: ignoringCharacters,
-                appending: &resultStringView)
+}
 
-            return resultStringView
-        }
-        catch {
-            return nil
-        }
-    }
+extension BankCardNumberUtils {
 
-    static func stringViewAndRange(
-        from originalStringView: String.UnicodeScalarView,
-        range originalRange: Range<String.UnicodeScalarIndex>,
-        preservingCharacters: CharacterSet,
-        ignoringCharacters: CharacterSet
-        ) -> (String.UnicodeScalarView, Range<String.UnicodeScalarIndex>)? {
+    static func cardNumberStringView(
+        fromDigits digitsStringView: String.UnicodeScalarView,
+        withLength digitsStringViewLength: Int,
+        sortedSpacesPositions: [Int]) -> String.UnicodeScalarView {
```

Comment:
Sorted [Int]? Like... IndexSet?

I mean, IndexSet provides some good underhood features like uniqueness of index and sorting by value.